src/ballet/chacha20/fd_chacha20_rng.c

```diff
@@ -0,0 +1,33 @@
+#include "fd_chacha20.h"
+
+static EVP_CIPHER_CTX *ctx = NULL;
```

Comment:
Firedancer doesn't have much static state like this. Prefer the footprint/align/alloc/new pattern that the bulk of the code follows. If that doesn't work in this situation, this needs a big comment block explaining why.

Reply:
totally agree, deleted this static state ✅, now we generates this EVP_CIPHER_CTX per call and free it from memory once it has finished